nuxeo-management-rest-api/src/test/java/org/nuxeo/rest/management/TestElasticsearchObject.java

```diff
@@ -0,0 +1,154 @@
+
+        // Create new documents without indexing them
+        createDocuments();
+
+        // Wait for an eventual indexation
+        waitForIndexing();
+
+        // Nothing indexed because of disable indexing flag
+        assertEquals(0, ess.query(new NxQueryBuilder(coreSession).nxql(GET_DOCUMENTS_QUERY)).totalSize());
+
+        // Starts the ES indexing
+        String path = String.format("site/management/elasticsearch/%s/reindex", coreSession.getRepositoryName());
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+            waitForIndexing();
+            assertEquals(2 + initialDocCount,
+                    ess.query(new NxQueryBuilder(coreSession).nxql(GET_DOCUMENTS_QUERY)).totalSize());
+        }
+    }
+
+    @Test
+    public void shouldFailRunningIndexationWhenRepositoryNotExists() {
+        // Launch the ES indexing
+        String path = String.format("site/management/elasticsearch/%s/reindex", "unExistingRepository");
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            assertEquals(SC_NOT_FOUND, response.getStatus());
+        }
+    }
+
+    @Test
+    public void shouldRunFlush() {
+        try (CloseableClientResponse response = httpClientRule.post("site/management/elasticsearch/flush", null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+        }
+    }
+
+    @Test
+    public void shouldRunFlushOnRepository() {
+        String path = String.format("site/management/elasticsearch/%s/flush", coreSession.getRepositoryName());
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+        }
+    }
+
+    @Test
+    public void shouldRunOptimize() {
+        try (CloseableClientResponse response = httpClientRule.post("site/management/elasticsearch/optimize", null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+        }
+    }
+
+    @Test
+    public void shouldRunOptimizeOnRepository() {
+        String path = String.format("site/management/elasticsearch/%s/optimize", coreSession.getRepositoryName());
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+        }
+    }
+
+    protected void createDocuments() {
+        Stream.of("my-file-1", "my-file-2").forEach((fileName) -> {
+            DocumentModel doc = coreSession.createDocumentModel("/default-domain/workspaces/", fileName, "File");
+            doc.setPropertyValue("dc:title", String.format("Title of %s", fileName));
+            doc.putContextData(ElasticSearchConstants.DISABLE_AUTO_INDEXING, Boolean.TRUE);
+            coreSession.createDocument(doc);
+        });
+
+        txFeature.nextTransaction();
```

Comment:
Sorry, I still find it not consistent to handle the transaction here, especially that after the call to `createDocuments(` you are calling `waitForIndexing(` that calls `txFeature.nextTransaction()`.
I think this call can just be removed here.